### PR TITLE
types: include `nuxt-link` target types

### PR DESCRIPTION
**Pull request by vencho-mdp**
### 🔗 Linked issue

resolves #8171 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Added possible types for target value, in order to provide autocompletion.

### 📝 Checklist
- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-link.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export type NuxtLinkProps = {
   custom?: boolean
 
   // Attributes
-  target?: string | null
+  target?: '_blank' | '_parent' | '_self' | '_top' | string
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-link.ts:29]:
```suggestion
  target?: '_blank' | '_parent' | '_self' | '_top' | (string & {}) | null
```

Comment (DamianGlowala) [reply packages/nuxt/src/app/components/nuxt-link.ts]: Knowing that `target` is optional, is `null` actually needed?

Comment (vencho-mdp) [reply packages/nuxt/src/app/components/nuxt-link.ts]: If null is a type option then it'd allow something like `:target="null"`. Not so sure if it has a use case.

Comment (DamianGlowala) [reply packages/nuxt/src/app/components/nuxt-link.ts]: Even if we wanted to use a ternary operator like the one below, I'd naturally go for `undefined` (knowing the property is optional).

```
:target="variable ? '_blank' : undefined"
```

@danielroe, shall all `null` values be removed for optional properties (line 14, 29 and 30)? There is some inconsistency at the moment.

Comment (danielroe) [reply packages/nuxt/src/app/components/nuxt-link.ts]: No, null values serve an important purpose of overwriting default values. See https://github.com/nuxt/framework/pull/3544#issuecomment-1062061430 and other comments there.

**Review by danielroe — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks <3

**Pull request merged**